shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -461,19 +474,43 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
   }
 }
 
-- (void)dispatchKeyEvent:(NSEvent*)event ofType:(NSString*)type {
+- (void)propagateKeyEvent:(NSEvent*)event {
+  for (NSResponder* responder in self.additionalKeyResponders) {
+    if (event.type == NSEventTypeKeyDown) {
+      if ([responder respondsToSelector:@selector(keyDown:)]) {
+        [responder keyDown:event];
+      }
+    } else {
+      if ([responder respondsToSelector:@selector(keyUp:)]) {
+        [responder keyUp:event];
+      }
+    }
+  }
+}
+
+- (void)dispatchKeyEvent:(NSEvent*)event {
+  if (event.type != NSEventTypeKeyDown && event.type != NSEventTypeKeyUp) {
+    return;
+  }
   NSMutableDictionary* keyMessage = [@{
     @"keymap" : @"macos",
-    @"type" : type,
+    @"type" : event.type == NSEventTypeKeyDown ? @"keydown" : @"keyup",
     @"keyCode" : @(event.keyCode),
     @"modifiers" : @(event.modifierFlags),
   } mutableCopy];
-  // Calling these methods on any other type of event will raise an exception.
-  if (event.type == NSEventTypeKeyDown || event.type == NSEventTypeKeyUp) {
-    keyMessage[@"characters"] = event.characters;
-    keyMessage[@"charactersIgnoringModifiers"] = event.charactersIgnoringModifiers;
-  }
-  [_keyEventChannel sendMessage:keyMessage];
+  keyMessage[@"characters"] = event.characters;
+  keyMessage[@"charactersIgnoringModifiers"] = event.charactersIgnoringModifiers;
+  __weak __typeof__(self) weakSelf = self;
+  FlutterReply replyHandler = ^(id _Nullable reply) {
+    if (reply == nil) {
```

Comment:
I believe the prevailing style in the macOS embedding is `if (!reply)`, but maybe I'm misremembering.

Reply:
Changed. I did some quick searches, but I don't see it either way, actually.